db_test.go

```diff
@@ -215,6 +215,79 @@ func TestOpen_ErrChecksum(t *testing.T) {
 	}
 }
 
+// Ensure that it can read the page size from the second meta page if the first one is invalid.
+// The page size is expected to be the OS's page size in this case.
+func TestOpen_ReadPageSize_FromMeta1_OS(t *testing.T) {
+	// Create empty database.
+	db := MustOpenDB()
+	path := db.Path()
+	defer db.MustClose()
+
+	// Close database.
+	if err := db.DB.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	// Read data file.
+	buf, err := ioutil.ReadFile(path)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Rewrite first meta page.
+	meta0 := (*meta)(unsafe.Pointer(&buf[pageHeaderSize]))
+	meta0.pgid++
+	if err := ioutil.WriteFile(path, buf, 0666); err != nil {
+		t.Fatal(err)
+	}
+
+	// Reopen data file.
+	if db, err := bolt.Open(path, 0666, nil); err != nil {
+		t.Fatalf("unexpected error: %s", err)
+	} else {
+		if db.Info().PageSize != os.Getpagesize() {
```

Comment:
I think we should have a test that does the same for custom-sized page, 
such that operating system page does not matter if the DB claims to be 8MB on the second page. 

In particular we should test a situation when we validate 8MB page populated from 1024bytes buffer.